src/classes/CacheProvider.js

```diff
@@ -0,0 +1,17 @@
+import { default as BaseProvider } from './BaseProvider';
+
+/**
+* Abstract class the provides a reference to an adapter instance (a caching implementation)
+*
+* @class
+* @extends {BaseProvider}
+* @param {Object} adapter - A cache adapter
+*/
+export class CacheProvider extends BaseProvider {
+  constructor() {
+    super(...arguments);
+    this.DEFAULT_ADAPTER = './MemoryCache';
```

Comment:
And also, I believe that it can't hurt to keep a hash map of appId -> provider in the BaseProvider, the same way the DB provider has it already, then if a provider is truly global, one can ignore the appID but by default it's supported, as is cache, DB etc...